src/main/java/appeng/tile/storage/TileDrive.java

```diff
@@ -349,4 +357,78 @@ public void blinkCell(final int slot) {
     public void saveChanges(final IMEInventory cellInventory) {
         this.worldObj.markTileEntityChunkModified(this.xCoord, this.yCoord, this.zCoord, this);
     }
+
+    public void lockCells() {
+        int idx = 0;
+        for (ICellHandler cellHandler : this.handlersBySlot) {
+            final ItemStack cell = this.inv.getStackInSlot(idx);
+            if (cellHandler == null || cell == null
+                    || !(cell.getItem() instanceof ItemExtremeStorageCell)
+                    || (cell.getItem() instanceof ItemExtremeStorageCell exCell && exCell.getTotalTypes(cell) != 1)) {
+                idx++;
+                continue;
+            }
+            final IMEInventoryHandler<?> inv = cellHandler.getCellInventory(cell, this, StorageChannel.ITEMS);
+            if (inv instanceof ICellInventoryHandler handler) {
+                final ICellInventory cellInventory = handler.getCellInv();
+                if (cellInventory != null) {
+                    if (cellInventory.getStoredItemTypes() != 0) {
+                        cellInventory.getConfigInventory().setInventorySlotContents(
+                                0,
+                                handler.getAvailableItems(new ItemList()).getFirstItem().getItemStack());
+                    }
+                }
+            }
+            idx++;
+        }
+
+    }
+
+    public boolean applyStickyToCells(EntityPlayer p) {
+        int idx = 0;
+        for (ICellHandler cellHandler : this.handlersBySlot) {
```

Comment:
Use normal for loop instead

Reply:
Resolved, see d64a646669a6487c86b053665044dae45a2d05be